src/main/java/net/minecraftforge/fml/common/registry/VillagerRegistry.java

```diff
@@ -269,6 +281,31 @@ private void register(VillagerCareer career)
             career.id = careers.size();
             careers.add(career);
         }
+
+        private void setId(int id)
+        {
+            if(id < nextId)
```

Comment:
I was using this to keep track of the number added, as well as to keep the vanilla compatibility, since vanilla was using int ids for everything , I guess I could instead check the number of professions registered, then just use registry lookups instead.